docs/builder/ref.md

```diff
@@ -0,0 +1,290 @@
+# Builder Library Reference
+
+## Top-Level Program Structure
+
+Here's the general structure of a program that uses the builder to generate Calyx code.
+
+```python
+# import the builder library
+import calyx.builder as cb
+
+
+# define my_subcomponent
+def add_my_subcomponent(prog):
+    # subcomponent definition here
+
+
+# define my_component
+def my_component(prog, my_subcomponent): 
+    # add the component to the program
+    my_component = prog.component("my_component")
+
+    # add my_subcomponent as a cell of my_component
+    my_subcomponent = my_component.cell("my_subcomponent", my_subcomponent)
```

Comment:
maybe something like `my_other_component`? or `second_component`.

And then change the name of the instance to something else to avoid muddying the distinction between an instance and a definition in the builder

Reply:
ah thank you both! to be clear, the LHS is a definition, and the RHS is an instance?

Reply:
if `LHS = .cell(STR, COMP` then `LHS` is just a handle, `STR` is the name of an instance and `COMP` is the definition (or really handle to one)

Reply:
thank you!!